source/extensions/filters/http/cache/cache_custom_headers.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "envoy/http/header_map.h"
+
+#include "common/http/headers.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+struct CacheCustomHeaders {
+  static CacheCustomHeaders& get();
+  CacheCustomHeaders();
+
+  Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::RequestHeaders>
+      authorization;
```

Comment:
I may need to go and page back in all of this code because I'm rusty on it, but I'm confused how this works. How do you guarantee that get() is called for the first time before the header map is finalized? If that is guaranteed somehow can you add more comments? If this works I agree it's better than requiring static init order to work.

Reply:
The CacheCustomHeaders struct gets instantiated during global static variable initialization, not during `get()`. I patched this in to the environment where I encountered the original RELEASE_ASSERT to ensure the refactor doesn't cause the problem you describe, cause I wasn't fully sure either.

If we used local static initialization within `get()`, to lazily initialize like my previous PR did, then we'd have the problem of registering after finalize, but, as currently written, it's ok as long as we use `CustomHeaderValues` and not `HeaderValues` as we are still registering far before the bootstrap has a chance to run.